File: backend/app/schemas/referral.py
from pydantic import BaseModel, Field


class ReferralCreate(BaseModel):
    job_id: str
    referrer_id: str
    message: str | None = Field(None, max_length=2000)


class ReferralUpdate(BaseModel):
    status: str = Field(..., pattern="^(accepted|rejected)$")


class ReferralResponse(BaseModel):
    id: str
    job_id: str
    requester_id: str
    referrer_id: str
    status: str
    message: str | None = None
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class ReferralListResponse(BaseModel):
    sent: list[ReferralResponse] = []
    received: list[ReferralResponse] = []

File: frontend/src/types/referral.ts
export interface Referral {
  id: string;
  job_id: string;
  requester_id: string;
  referrer_id: string;
  status: "pending" | "accepted" | "rejected";
  message: string | null;
  created_at: string;
  updated_at: string;
}

export interface ReferralCreate {
  job_id: string;
  referrer_id: string;
  message?: string;
}

export interface ReferralListResponse {
  sent: Referral[];
  received: Referral[];
}
